specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,264 @@
+# Background
+
+We are exposing an event that will fire when an attempt to launch a registered protocol is made. The host will be given the option to cancel the launch, handle the popup dialog to hide the dialog, as well as revoke permissions that have been previously given to the origin launching the protocol. 
+
+# Description
+
+This event will fire before the registered protocol launch occurs. Currently a popup dialog is displayed in which the user can click `Open` or `Cancel`. If the request is made from a trustworthy origin a checkmark box will be displayed that will allow the user to always allow this registered protocol from this origin. The `NavigationStarting`, `NavigationCompleted`, `SourceChanged`, `ContentLoading`, and `HistoryChanged` events will not fire when a request is made to launch a registered protocol. 
+
+There are two events associated with the registered protocol launch - one for the main frame, and one for non-main frame(s). In the case in which the launch request is made from a non-main frame, the frame will raise a `LaunchingRegisteredProtocol` event as well as `CoreWebView2.FrameLaunchingRegisteredProtocol` event. 
+# Examples
+
+## Win32 C++
+    
+```cpp 
+
+CHECK_FAILURE(m_webView->add_LaunchingRegisteredProtocol(
+    Callback<ICoreWebView2LaunchingRegisteredProtocolEventHandler>(
+        [this](
+            ICoreWebView2* sender,
+            ICoreWebView2LaunchingRegisteredProtocolEventArgs* args) {
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(args->get_Uri(&uri));
+                if (wcscmp(uri.get(), L"calculator://") == 0)
+                {
+                    CHECK_FAILURE(args->put_Handled(TRUE));
+                    // If this matches our desired protocol, then suppress the
+                    // popup dialog.
+                }
+                else
+                {
+                    CHECK_FAILURE(args->put_Handled(FALSE));
```

Comment:
FALSE is the default value for Handled right? You shouldn't need this else block. But would be good to keep the 'Otherwise allow...' comment